### PR TITLE
MIP-B21: MetaMorpho Vault Activation

### PR DESCRIPTION
**Pull request by ElliotFriedman**
MIP-B21: MetaMorpho Vault Activation.

This PR creates a governance proposal that if successful executes the following actions on Base from the Temporal Governor:
1.  Accept ownership of the Moonwell USDC Metamorpho Vault
2. Accept ownership of the Moonwell WETH Metamorpho Vault
3. Transfer 50m WELL from the Foundation Base Multisig to the Moonwell Metamorpho URD

### CODE REVIEW

File: src/proposals/mips/mip-b21/mip-b21.sol [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+        );
+        _pushCrossChainAction(
+            addresses.getAddress("WETH_METAMORPHO_VAULT"),
+            abi.encodeWithSignature("acceptOwnership()"),
+            "Accept ownership of the Moonwell WETH Metamorpho Vault"
+        );
+        _pushCrossChainAction(
+            addresses.getAddress("xWELL_PROXY"),
+            abi.encodeWithSignature(
+                "transferFrom(address,address,uint256)",
```
Comment (MerlinEgalite) [inline src/proposals/mips/mip-b21/mip-b21.sol:100]: when did the approval occur? Is the `FOUNDATION_MULTISIG` the sender of the tx?

Comment (lyoungblood) [reply src/proposals/mips/mip-b21/mip-b21.sol:100]: @MerlinEgalite after the KYC/KYB process is completed by Morpho and the Moonwell Foundation directors approve the grant, the approval will be opened in order for the proposal to properly execute.

File: utils/Addresses.json [hunk 2]
```diff
@@ -299,6 +299,12 @@
         "name": "USDBC",
         "isContract": true
     },
+    {
+        "addr": "0x9e3380f8B29E8f85cA19EFFA80Fb41149417D943",
```
Comment (lyoungblood) [inline utils/Addresses.json:303]: Confirmed this is the correct URD provided by Morpho with owner [0x937Ce2d6c488b361825D2DB5e8A70e26d48afEd5](https://basescan.org/address/0x937Ce2d6c488b361825D2DB5e8A70e26d48afEd5)

File: utils/Addresses.json [hunk 3]
```diff
@@ -309,7 +315,7 @@
         "addr": "0xd5F9BaE72F6D8334ac4BB6C14463A81A43037E1C",
         "chainId": 8453,
         "isContract": true,
-        "name": "ETH_METAMORPHO_FEE_SPLITTER"
+        "name": "WETH_METAMORPHO_FEE_SPLITTER"
     },
     {
         "addr": "0xc1256Ae5FF1cf2719D4937adb3bbCCab2E00A2Ca",
```
Comment (lyoungblood) [inline utils/Addresses.json:321]: Confirmed this is the correct mwUSDC vault.
- `pendingOwner` is currently set to the 0 address, but should be set to the [Moonwell Temporal Governor](https://basescan.org/address/0x8b621804a7637b781e2BbD58e256a591F2dF7d51) at `0x8b621804a7637b781e2BbD58e256a591F2dF7d51`
- `pendingGuardian` is currently set to the 0 address, but should be set to the Base Pause Guardian multisig at [0xB9d4acf113a423Bc4A64110B8738a52E51C2AB38](https://app.safe.global/home?safe=base:0xB9d4acf113a423Bc4A64110B8738a52E51C2AB38)
- `skimRecipient` is currently set to the 0 address, but should be set to the `USDC_METAMORPHO_FEE_SPLITTER` at [0x1e1b4a620d1a840bcd56706d87903e3583c8b26f]https://basescan.org/address/0x1e1b4a620d1a840bcd56706d87903e3583c8b26f
- `timelock` is currently set to 86400 and should be increased to 345600 after the updates above are made.

File: utils/Addresses.json [hunk 4]
```diff
@@ -321,7 +327,7 @@
         "addr": "0xa0E430870c4604CcfC7B38Ca7845B1FF653D0ff1",
         "chainId": 8453,
         "isContract": true,
-        "name": "ETH_METAMORPHO_VAULT"
+        "name": "WETH_METAMORPHO_VAULT"
     },
     {
         "addr": "0x833589fCD6eDb6E08f4c7C32D4f71b54bdA02913",
```
Comment (lyoungblood) [inline utils/Addresses.json:333]: Confirmed this is the correct mwETH vault.
- `pendingOwner` is currently set to the 0 address, but should be set to the [Moonwell Temporal Governor](https://basescan.org/address/0x8b621804a7637b781e2BbD58e256a591F2dF7d51) at `0x8b621804a7637b781e2BbD58e256a591F2dF7d51`
- `pendingGuardian` is currently set to the 0 address, but should be set to the Base Pause Guardian multisig at [0xB9d4acf113a423Bc4A64110B8738a52E51C2AB38](https://app.safe.global/home?safe=base:0xB9d4acf113a423Bc4A64110B8738a52E51C2AB38)
- `skimRecipient` is currently set to the 0 address, but should be set to the `WETH_METAMORPHO_FEE_SPLITTER` at [0xd5F9BaE72F6D8334ac4BB6C14463A81A43037E1C](https://basescan.org/address/0xd5F9BaE72F6D8334ac4BB6C14463A81A43037E1C)
- timelock is currently set to `86400` and should be increased to `345600` after the updates above are made.

File: utils/Addresses.json [hunk 5]
```diff
@@ -299,6 +299,12 @@
...
         "isContract": true
     },
+    {
+        "addr": "0x9e3380f8B29E8f85cA19EFFA80Fb41149417D943",
+        "chainId": 8453,
+        "isContract": true,
+        "name": "MOONWELL_METAMORPHO_URD"
+    },
     {
         "addr": "0x1e1b4a620d1A840bcD56706d87903e3583c8B26f",
```
Comment (lyoungblood) [inline utils/Addresses.json:309]: Confirmed this is the correct USDC fee splitter:
- `SPLIT_TOTAL` is 10,000, `splitA` and `splitB` are both 5,000 each, which is correct
- `metaMorphoVault` has the correct address: [0xc1256Ae5FF1cf2719D4937adb3bbCCab2E00A2Ca](https://basescan.org/address/0xc1256Ae5FF1cf2719D4937adb3bbCCab2E00A2Ca)
- `token` has the correct address of the USDC token on Base: [0x833589fCD6eDb6E08f4c7C32D4f71b54bdA02913](https://basescan.org/address/0x833589fCD6eDb6E08f4c7C32D4f71b54bdA02913)
- `mToken` has the correct address of the mUSDC mToken on Base: [0xEdc817A28E8B93B03976FBd4a3dDBc9f7D176c22](https://basescan.org/address/0xEdc817A28E8B93B03976FBd4a3dDBc9f7D176c22)

File: utils/Addresses.json [hunk 6]
```diff
@@ -309,7 +315,7 @@
         "addr": "0xd5F9BaE72F6D8334ac4BB6C14463A81A43037E1C",
```
Comment (lyoungblood) [inline utils/Addresses.json:315]: Confirmed this is the correct ETH fee splitter:
- `SPLIT_TOTAL` is 10,000, `splitA` and `splitB` are both 5,000 each, which is correct
- `metaMorphoVault` has the correct address: [0xa0E430870c4604CcfC7B38Ca7845B1FF653D0ff1](https://basescan.org/address/0xa0E430870c4604CcfC7B38Ca7845B1FF653D0ff1)
- `token` has the correct address of the WETH token on Base: [0x4200000000000000000000000000000000000006](https://basescan.org/address/0x4200000000000000000000000000000000000006)
- `mToken` has the correct address of the mWETH mToken on Base: [0x628ff693426583D9a7FB391E54366292F509D457](https://basescan.org/address/0x628ff693426583D9a7FB391E54366292F509D457)

**Review by lyoungblood — CHANGES_REQUESTED** [review]
My review focused on contract address and parameter accuracy rather than the Solidity code itself.

The proposal looks good from that perspective, however, the BlockAnalitica team will need to update the contract parameters on the two vaults as indicated in my comments before the governance proposal is executed, so I'm marking it "Request changes" (even though there are likely to be no changes to the PR itself).

File: src/proposals/mips/mip-b21/mip-b21.sol (hunk 1 above)
Comment (MerlinEgalite) [reply src/proposals/mips/mip-b21/mip-b21.sol:100]: very clear thx

**Review by lyoungblood — APPROVED** [review]

**Pull request merged**